src/ballet/pack/fd_est_tbl.h

```diff
@@ -0,0 +1,181 @@
+     bin */
+  double x;
+  /* x2: The numerator of the EMA of the square of values that have mapped
+     to this bin */
+  double x2;
+  /* d: The denominator for EMA(x), paired with the numerator from above.
+     */
+  double d;
+  double d2;
+};
+typedef struct fd_private_est_tbl_bin fd_est_tbl_bin_t;
+
+/* The main data structure described in the overall header comment */
+struct __attribute__((aligned(FD_EST_TBL_ALIGN))) fd_private_est_tbl {
+  /* magic: set to FD_EST_TBL_MAGIC */
+  ulong  magic;
+  /* bin_cnt_mask: (bin_cnt_mask+1) is the number of bins in the table, a power
+     of two */
+  ulong  bin_cnt_mask;
+  /* ema_coeff: the decay coefficient used in EMA computations. Near 1.0. */
+  double ema_coeff;
+  /* default_val: the value to return as mean when the query maps to a bin with
+     very few values */
+  double default_val;
+  /* 32 byte aligned at this point */
+  /* bins: the array of (bin_cnt_mask+1) bins follows.  The array size of 1 is
+     just convention. */
+  fd_est_tbl_bin_t bins[1];
+};
+typedef struct fd_private_est_tbl fd_est_tbl_t;
+
+
+FD_PROTOTYPES_BEGIN
+/* fd_est_tbl_{align, footprint} given the needed alignment and footprint for a
+   memory region suitable to hold fd_est_tbl's state.  bin_cnt specifies the
+   number of bins that the estimation table stores.  bin_cnt must be a
+   power-of-two greater than 0.  Increasing the number of bins increases the
+   footprint requirements but also increases the accuracy slightly (by reducing
+   collisions).  fd_est_tbl_{align, footprint} return the same value as
+   FD_EST_TBL_{ALIGN, FOOTPRINT}.
+
+   fd_est_tbl_new takes ownership of the memory region pointed to by mem (which
+   is assumed to be non-NULL and have the appropriate alignment and footprint)
+   and formats it as a fd_est_tbl.  The estimation table will use bin_cnt bins,
+   and each bin's EMA will be tuned for an a window size of history.  history
+   must be positive.  The table will use a default value of default_val for the
+   mean whenever a query indicates a bin has had no data.  Returns mem (which
+   will be formatted for use) on success and NULL on failure (bad inputs).  The
+   caller will not be joined to the region on return.
+
+   fd_est_tbl_join joins the caller to a memory region holding the state of a
+   fd_est_tbl.
+
+   fd_est_tbl_leave leaves the current join.  Returns a pointer in the local
+   address space to the memory region holding the table state.  The join should
+   not be used after calling _leave.
+
+   fd_est_tbl_delete unformats the memory region used to hold the state of an
+   fd_est_tbl and returns ownership of the underlying memory region to the
+   caller.  There should be no joins in the system on the fd_est_tbl.  Returns
+   a pointer to the underlying memory region. */
+
+FD_FN_CONST static inline ulong fd_est_tbl_align    ( void ) { return FD_EST_TBL_ALIGN; }
+FD_FN_CONST static inline ulong fd_est_tbl_footprint( ulong bin_cnt ) {
```

Comment:
Can you make this a `FD_LAYOUT` macro, eg like `FD_FCTL_FOOTPRINT`

https://github.com/firedancer-io/firedancer/blob/766a67ae138e79e922e02d679204de0cc513e6bb/src/tango/fctl/fd_fctl.h#L24